Source/Client/Windows/HostWindow.cs

```diff
@@ -148,12 +145,15 @@ public override void DoWindowContents(Rect inRect)
             var buttonRect = new Rect((inRect.width - 100f) / 2f, inRect.height - 35f, 100f, 35f);
 
             if (Widgets.ButtonText(buttonRect, "MpHostButton".Translate()))
+            {
+                LoadedModManager.GetMod<MultiplayerMod>().WriteSettings();
                 TryHost();
+            }
         }
 
         private void TryHost()
         {
-            if (direct && !TryParseIp(addressBuffer, out settings.bindAddress, out settings.bindPort))
+            if (settings.direct && !TryParseIp(settings.directAddress, out settings.bindAddress, out settings.bindPort))
```

Comment:
Why are we not parsing addressBuffer but directAddress? Did I miss something above?

Reply:
Other than this it looks fine, can you also provide documentation on the code you changed so other people will know how to use it?